### PR TITLE
chore: add multirm router layer to rm module

### PR DESCRIPTION
**Pull request by carolinaecalderon**
## Description
Add the multirm module back in (from #8857 ) , but look up the right resource manager from the globally unique resource pool name provided.

Also delete sproto messages that wrap under 3 variables, replace this directly with the variables in the RM interface. This reduces bloat in the repo, as sproto message wrappers are remnants of the actor system.

## Test Plan
See multirm_intg_test.go

## Commentary (optional)
Since I'm making changes to the signatures of methods in the RM interface, there are a lot of one-line changes in the repo. I recommend looking at the `resource_manager_iface.go`, and the `multirm` module.
There's no check if there are multiple same-named resource pools at this level -- that will be the responsibility of the config to check

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket
RM-73

### CODE REVIEW

File: master/internal/rm/resource_manager_iface.go [hunk 1]
```diff
@@ -39,14 +37,14 @@ type ResourceManager interface {
 	) (model.TaskContainerDefaultsConfig, error)
 
 	// Job queue
-	GetJobQ(sproto.GetJobQ) (map[model.JobID]*sproto.RMJobInfo, error)
+	GetJobQ(string) (map[model.JobID]*sproto.RMJobInfo, error)
```
Comment (stoksc) [inline master/internal/rm/resource_manager_iface.go:40]: i'd like for us to type resource pools at some point

Comment (stoksc) [reply master/internal/rm/resource_manager_iface.go:40]: (`type ResourcePoolName string`)

File: master/internal/rm/multirm/multirm.go [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/proto/pkg/apiv1"
+	"github.com/determined-ai/determined/proto/pkg/jobv1"
+)
+
+// ErrRPNotDefined returns a detailed error if a resource pool isn't found.
+func ErrRPNotDefined(rp string) error {
+	return fmt.Errorf("could not find resource pool %s", rp)
+}
+
```
Comment (stoksc) [inline master/internal/rm/multirm/multirm.go:23]: i'm trusting this is unchanged since it was last landed

**Review by stoksc — APPROVED** [review]
lgtm

File: master/internal/rm/multirm/multirm.go (hunk 2 above)
Comment (carolinaecalderon) [reply master/internal/rm/multirm/multirm.go:23]: I had to change the signatures to _not_ take in a RM name, and similarly, in getRM, I removed references to RM name

**Pull request merged**